dockstore/dockstore-ui2
# Metrics tab changes for new validation metrics schema

**Description**
This PR https://github.com/dockstore/dockstore/pull/5486 changed the DB schema for the validation status metric which required these front-end changes. As a result of the schema change, each validator tool version now stores more info, so I made changes in the validation table to display this info.

Changes:
- Created two mat cards, one for Execution Metrics and one for Validator Tool Metrics, to differentiate between the two types of metrics
  - As a result, I renamed the tab from Executions to Metrics
- Added a preview banner for the new feature
- Rounded double values to two decimal places. The full value can be viewed in the mat tooltip.
- The validator tool metrics card has a drop down for each validator tool. The table shows info for each validator tool version.
- The default platform value when navigating to the Metrics tab is `ALL` for all platforms

**Before**: View this PR https://github.com/dockstore/dockstore-ui2/pull/1758 for more before screenshots
![image](https://user-images.githubusercontent.com/25287123/235706974-206acc8c-ad8a-49e2-96f7-e7f9c15f0e9a.png)

**After:**
Execution and Validator Tool metrics exist:
![Screenshot 2023-05-02 at 09-12-14 Dockstore Workflow github com_gatk-workflows_seq-format-conversion_BAM-to-Unmapped-BAM](https://user-images.githubusercontent.com/25287123/235704822-8e03e2f2-0c4c-4cfd-95ab-f988856756d9.png)

Only Validator Tool metrics exist:
![Screenshot 2023-05-02 at 09-12-34 Dockstore Workflow github com_gatk-workflows_seq-format-conversion_BAM-to-Unmapped-BAM](https://user-images.githubusercontent.com/25287123/235704842-74a927d0-f8ff-41fd-b487-5c6c7a87b2f4.png)

Execution metrics exist, except only Execution Status metrics are present (no CPU, Memory, or Execution Time metrics):
![Screenshot 2023-05-02 at 09-12-44 Dockstore Workflow github com_gatk-workflows_seq-format-conversion_BAM-to-Unmapped-BAM](https://user-images.githubusercontent.com/25287123/235704859-8a303e86-f0ef-4ec2-a00c-ea4480b3ac52.png)

No metrics exist:
![Screenshot 2023-05-02 at 09-13-08 Dockstore Workflow github com_gatk-workflows_seq-format-conversion_BAM-to-Unmapped-BAM](https://user-images.githubusercontent.com/25287123/235704905-ae18dc9a-c14f-4942-95ee-14af664fbc59.png)

Video:

https://user-images.githubusercontent.com/25287123/235710571-6515068b-c6dc-4238-8838-f0a03f08679f.mp4


**Review Instructions**
- Go to a workflow version with metrics and add the `metrics` feature flag to the URL
- Click on the Metrics tab
- There should be a preview banner
- Click through the Platform drop down and verify that the tables look sane

**Issue**
[SEAB-5319](https://ucsc-cgl.atlassian.net/browse/SEAB-5319)

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!



[SEAB-5319]: https://ucsc-cgl.atlassian.net/browse/SEAB-5319?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
